finance-api/internal/service/invoices.go

```
@@ -0,0 +1,88 @@
+package service
+
+import (
+	"context"
+	"github.com/jackc/pgx/v5/pgtype"
+	"github.com/opg-sirius-finance-hub/shared"
+)
+
+func (s *Service) GetInvoices(clientID int) (*shared.Invoices, error) {
+	ctx := context.Background()
+
+	var invoices shared.Invoices
+
+	invoicesRawData, err := s.Store.GetInvoices(ctx, pgtype.Int4{Int32: int32(clientID), Valid: true})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, i := range invoicesRawData {
+		ledgerAllocations, totalOfLedgerAllocationsAmount, err := getLedgerAllocations(s, ctx, int(i.ID))
+		if err != nil {
+			return nil, err
+		}
+		var invoice = shared.Invoice{
+			Id:                 int(i.ID),
+			Ref:                i.Reference,
+			Status:             "",
+			Amount:             int(i.Amount),
+			RaisedDate:         shared.Date{Time: i.Raiseddate.Time},
+			Received:           totalOfLedgerAllocationsAmount,
+			OutstandingBalance: int(i.Amount) - totalOfLedgerAllocationsAmount,
+			Ledgers:            ledgerAllocations,
+			SupervisionLevels:  getSupervisionLevels(s, ctx, int(i.ID)),
+		}
+
+		invoices = append(invoices, invoice)
+	}
+
+	return &invoices, nil
+}
+
+func getSupervisionLevels(s *Service, ctx context.Context, invoiceID int) []shared.SupervisionLevel {
+	var supervisionLevels []shared.SupervisionLevel
+	supervisionLevelsRawData, err := s.Store.GetSupervisionLevels(ctx, pgtype.Int4{Int32: int32(invoiceID), Valid: true})
+	if err != nil {
+		return nil
```

Don't silently fail. This should also return an error if an error occurs.

---

done